### DISCUSSION TITLE
Inclusion of moorings?

### DISCUSSION THREAD

**Discussion by saltynexus** [capytaine/capytaine#407] [Q&A] [answered]
Does Capytaine support the inclusion of moorings? I ask because I believe ANSYS AQWA supports the inclusion of moorings and I'm looking for an alternative to ANSYS AQWA. I know WEC-SIM couples with MoorDyn, but I'm not to familiar with Capytaine. I tried to search the documentation and didn't find anything, which leads me to believe it doesn't. I suspect it's because Capytaine is a frequency domain model and, for example, MoorDyn is a time domain model? If there is not mooring support, is there future plans to include support? Thanks.

**Comment by mancellin** [comment, marked as answer]
Thank you for your question.

The goal of Capytaine is to compute interaction forces between floating bodies and the ocean. 
Simulating the dynamics of the body including other forces requires another tool (such as WEC-SIM), although some limited frequency domain evolution equations can be done with Capytaine (e.g. RAO computation).
(The scope of AQWA is maybe broader than the scope of Capytaine, it may include the time-domain simulation.)

Coupling with Moordyn is out of the scope of Capytaine. Adding some linear mooring model as a stiffness matrix in the RAO might be possible, but it is not on the roadmap.

**Reply by saltynexus** [reply]
OK...got ya. I guess I'm now wondering, does Capytaine do two body simulations?

**Reply by mancellin** [reply]
Yes, see e.g. this example from the cookbook: https://capytaine.github.io/stable/user_manual/cookbook.html#simulation-with-several-bodies

**Reply by saltynexus** [reply]
Forgive my ignorance, but if you can do several bodies, shouldn't you be able to include the influence of moorings as a "second body" in which the coupling occurs at the linkage? I guess I need to read up more on this subject. Any book recommendations?

**Reply by mancellin** [reply]
What I mean by two-body-simulation in Capytaine is computing the fluid flow and fluid-structure interaction in a configuration where several bodies are present.

Mooring modeling would require a model for the deformation of the cable (what e.g. Moordyn is doing) and a model for the cable-body joint (what e.g. WEC-SIM is doing). This is out of the scope of Capytaine which is basically a computational fluid dynamics code[^1].

[^1]: "CFD" usually refer to another kind of model, but taking the words literally Capytaine is about "computational fluid dynamics".
